File: Minimum Depth of Binary Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(root == null)
        {
            return 0;
        }
        if(root.left == null && root.right == null)
        {
            return 1;
        }
        
        int leftDepth = Integer.MAX_VALUE;
        int rightDepth = Integer.MAX_VALUE;
        if(root.left != null)
        {
            leftDepth = minDepth(root.left);
        }
        
        if(root.right != null)
        {
            rightDepth = minDepth(root.right);
        }
        
        int depth = Math.min(leftDepth, rightDepth);
        return depth + 1;
    }
}

File: Minimum Depth of Binary Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode *root) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        
        if(root==NULL)
            return 0;
        
        int depth = 0;
        
        if(root->left||root->right)
        {
            int left = INT_MAX;
            int right = INT_MAX;
            if(root->left)
            {
                left = minDepth(root->left);
            }
            if(root->right)
            {
                right = minDepth(root->right);
            }
            
            depth = min(left, right);
    
            
        }
        
        return depth+1;
    }
};